server_app/user/handler.js

```diff
@@ -95,6 +95,34 @@ const resetPassword = async(req, res) => {
   }
 }
 
+const updatePassword = async(req, res) => {
+  try {
+    const mail = /*decipher.decode(*/req.body.mail;
+    const password = /*decipher.decode(*/req.body.password;
+    const newPassword = /*decipher.decode(*/req.body.newpassword;
+    const updated = await userService.updatePassword(mail, password, newPassword);
+    return res.status(200).json({updated: updated});
+  } catch (error) {
+    console.error(error);
+    return res.status(500).send("Ocorreu um erro no tratamento do seu request!");
+  }
+}
+
+const updateUserData = async(req, res) => {
+  try {
+    const userId = /*decipher.decode(*/req.body.user_id;
+    const newMail = /*decipher.decode(*/req.body.mail;
+    const newUsername = /*decipher.decode(*/req.body.username;
+
+		const updated = await userService.updateUserData(userId, newUsername, newMail);
```

Comment:
I think we should validate the newMail in this layer before send it to the service. I have a[ helper validator ](https://github.com/brmodeloweb/brmodelo-app/blob/main/server_app/user/validator.js) that you can call an put your you validation